lib/inspec/rule.rb

```diff
@@ -332,7 +332,7 @@ def __apply_waivers
       input_name = @__rule_id # TODO: control ID slugging
       registry = Inspec::InputRegistry.instance
       input = registry.inputs_by_profile.dig(__profile_id, input_name)
-      return unless input
+      return unless input && input.value.is_a?(Hash)
```

Comment:
I can confirm that inputs can indeed have Hash values; but here they'd have to have a hash-valued Input with the same name as a control, which is increasingly unlikely. Has anyone asked the affected users if they use Hashes?

One note, input.value may throw a WARNing just by calling it if the input has no value; you may want to add:

`input && input.has_value? && input.value.is_a?(Hash)`

which will be quieter. #4995 addresses this a bit more.

Reply:
Thanks for this! :) 

I know that if you `pry` into this with a control that matches the affected user's, line 334:

`input = registry.inputs_by_profile.dig(__profile_id, input_name)`

Does return an `input` where `value` is the string `false`, not a hash. Which will break https://github.com/inspec/inspec/blob/7242da8bf557b6a76ac32d698a59d1d43a98ad04/lib/inspec/rule.rb#L343

I wasn't able to dig deep enough into the internals to figure out why clashing controls and input namespace does this, though I have some half-baked theories. I'll update per your suggestion & today writing a wider issue to look into this which I would love your feedback on! 👍

Reply:
p.s. I don't resolve to say the conversation is over, just to note that I've achieved the changes so feel free to reopen 👍